tests/test_hra.py

```diff
@@ -326,6 +326,41 @@ def test_criteria_table_parsing(self):
         pandas.testing.assert_frame_equal(
             expected_composite_dataframe, composite_dataframe)
 
+    def test_criteria_table_parsing_with_bom(self):
+        """HRA: criteria table - parse a BOM."""
+        from natcap.invest import hra
+
+        criteria_table_path = os.path.join(self.workspace_dir, 'criteria.csv')
+        with open(criteria_table_path, 'w', encoding='utf-8-sig') as criteria_table:
+            criteria_table.write(
+                textwrap.dedent(
+                    """\
+                    HABITAT NAME,eelgrass,,,hardbottom,,,CRITERIA TYPE
+                    HABITAT RESILIENCE ATTRIBUTES,RATING,DQ,WEIGHT,RATING,DQ,WEIGHT,E/C
+                    recruitment rate,2,2,2,2,2,2,C
+                    connectivity rate,2,2,2,2,2,2,C
+                    ,,,,,,,
+                    HABITAT STRESSOR OVERLAP PROPERTIES,,,,,,,
+                    oil,RATING,DQ,WEIGHT,RATING,DQ,WEIGHT,E/C
+                    frequency of disturbance,2,2,3,2,2,3,C
+                    management effectiveness,2,2,1,2,2,1,E
+                    ,,,,,,,
+                    fishing,RATING,DQ,WEIGHT,RATING,DQ,WEIGHT,E/C
+                    frequency of disturbance,2,2,3,2,2,3,C
+                    management effectiveness,2,2,1,2,2,1,E
+                    """
+                ))
+
+        # Sanity check: make sure the file has the expected BOM
+        bom_char = "\uFEFF"  # byte-order marker in 16-bit hex value
+        with open(criteria_table_path) as criteria_table:
+            assert criteria_table.read().startswith(bom_char)
```

Comment:
This bit is still failing on Windows and I think it's because of the following from Python's codec docs (https://docs.python.org/3/library/codecs.html#encodings-and-unicode)

>On encoding the utf-8-sig codec will write 0xef, 0xbb, 0xbf as the first three bytes to the file. On decoding utf-8-sig will skip those three bytes if they appear as the first three bytes in the file. In UTF-8, the use of the BOM is discouraged and should generally be avoided.

So maybe it's being stripped out when opening and decoding? Although this is contradictory to what I'm seeing in my terminal:

```python
>>> with open("test-bom.csv", 'w', encoding='utf-8-sig') as my_table:
...     my_table.write("HABITAT NAME,eelgrass,,,hardbottom,,,CRITERIA TYPE")
>>>
>>> with open("test-bom.csv") as table:
...     print(table.read())
...
ï»¿HABITAT NAME,eelgrass,,,hardbottom,,,CRITERIA TYPE
```
Which makes sense because:
>To increase the reliability with which a UTF-8 encoding can be detected, Microsoft invented a variant of UTF-8 (that Python calls "utf-8-sig") for its Notepad program: Before any of the Unicode characters is written to the file, a UTF-8 encoded BOM (which looks like this as a byte sequence: 0xef, 0xbb, 0xbf) is written. As it’s rather improbable that any charmap encoded file starts with these byte values (which would e.g. map to

>LATIN SMALL LETTER I WITH DIAERESIS
>RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK
>INVERTED QUESTION MARK

So, I'm not sure why the test assertion error looks like the BOM is being decoded and stripped, where as when I open and read it I get the funky character mapping...